src/bosh-director/lib/bosh/director/jobs/ssh.rb

```diff
@@ -21,17 +22,32 @@ def initialize(deployment_id, options = {})
 
       def perform
         target = Target.new(@target_payload)
-
         filter = {}
-        filter[:job] = target.job if target.job
-        filter.merge!(target.id_filter)
+        if target.job && ip_address?(target.job) && instance_count(target.job) == 0
+          ip_address = target.job
+          net_address = ip_to_netaddr(ip_address)
+          ip_error_message = "{:ip=>#{ip_address}}"
+          ipaddress_model = Models::IpAddress.where(address: net_address.to_i).first
+          if ipaddress_model.nil?
+            filter = []
+            filter << Sequel.like(:spec_json, "%\\\"ip\\\":\\\"#{ip_address}\\\"%")
```

Comment:
Hi Barthy1

We wouldn't expect to be aware of Sequel interface at this point -- can we move this parameter into a hash component in the call to `filter_by`, and rely on the InstanceLookup to implement the `spec_json` search in DB? This seems like a high level of exposure of the DB layout/contract.

what do you think?
eve

Reply:
Additionally, it looks like L#28-30 can be pushed inside the instance lookup also?